scripts/webpack.client.config.js

```diff
@@ -4,58 +4,103 @@ const merge = require("webpack-merge");
+const { GenerateSW } = require("workbox-webpack-plugin");
+
 // const { BundleAnalyzerPlugin } = require("webpack-bundle-analyzer");
 
 const root = path.join(__dirname, "..");
 const prod = process.env.NODE_ENV === "production" || process.env.CI === "true";
 
 module.exports = (options = {}) => merge(
-	require("./webpack.general.config")(options), {
-	devtool: prod ? "none" : "cheap-module-eval-source-map",
-	mode: prod ? "production" : "development",
-	entry: prod ? options.entry : [
-		"webpack-hot-middleware/client?reload=true&quiet=true",
-		options.entry,
-	],
-	module: {
-		rules: [{
-			test: /\.s?css$/,
-			// This is required otherwise it'll fail to resolve CSS in common.
-			include: root,
-			use: [{
-				loader: MiniCssExtractPlugin.loader,
-			}, {
-				loader: "css-loader",
-			}, {
-				loader: "sass-loader",
-			}],
-		}, {
-			test: /\.(png|ttf|woff|eot|woff2)$/,
-			use: [{
-				loader: "file-loader",
-				options: {
-					name: "[path][name].[ext]",
-				},
-			}],
-		}, {
-			test: /\.svg$/,
-			loader: 'url-loader'
-		}],
-	},
-	plugins: [
-		new MiniCssExtractPlugin({
-			filename: "[name].css",
-			chunkFilename: "[id].css",
-		}),
-		new HtmlWebpackPlugin({
-			template: options.template,
-		}),
-		new PreloadWebpackPlugin({
-			rel: "preload",
-			as: "script",
-		}),
-	].concat(prod ? [] : [
-		new webpack.HotModuleReplacementPlugin(),
-	]),
-	target: "web",
-});
+  require("./webpack.general.config")(options), {
+    devtool: prod ? "none" : "cheap-module-eval-source-map",
+    mode: prod ? "production" : "development",
+    entry: prod ? options.entry : [
+      "webpack-hot-middleware/client?reload=true&quiet=true",
+      options.entry,
+    ],
+    module: {
+      rules: [{
+        test: /\.s?css$/,
+        // This is required otherwise it'll fail to resolve CSS in common.
+        include: root,
+        use: [{
+          loader: MiniCssExtractPlugin.loader,
+        }, {
+          loader: "css-loader",
+        }, {
+          loader: "sass-loader",
+        }],
+      }, {
+        test: /\.(png|ttf|woff|eot|woff2)$/,
+        use: [{
+          loader: "file-loader",
+          options: {
+            name: "[path][name].[ext]",
+          },
+        }],
+      }, {
+        test: /\.svg$/,
+        loader: 'url-loader'
+      }],
+    },
+    plugins: [
+      new MiniCssExtractPlugin({
+        filename: "[name].css",
+        chunkFilename: "[id].css"
+      }),
+      new HtmlWebpackPlugin({
+        template: options.template
+      }),
+      new PreloadWebpackPlugin({
+        rel: "preload",
+        as: "script"
+      }),
+      new WebpackPwaManifest({
+        name: "Coder",
+        short_name: "Coder",
+        description: "Run VS Code on a remote server",
+        background_color: "#303030",
+        icons: [
+          {
+            src: path.resolve("./assets/logo.png"),
+            sizes: [96, 128, 192, 256, 384]
+          }
+        ]
+      }),
+      new GenerateSW({
```

Comment:
Would it make sense to only use the service worker for production? I'm running into a few odd issues while developing using `yarn start`. Both Chromium and Firefox stop refreshing automatically at some point, and Firefox does some really weird things like refreshing infinitely or refusing to connect to the web socket. Is anyone else experiencing these issues? I think it might have something to do with how the service worker mixes with hot module reloading.

Everything seems to run beautifully when running the production-built binary.

Reply:
I agree. It used to be in the prod only array, but that didn't work properly for me in the docker container, but I presume I just built it wrong then. Ill move it back again.

Reply:
Ah yeah, not your fault, we didn't have `NODE_ENV` set in the Dockerfile.